qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/codeaction/QuteCodeActionForSimilarTextSuggestionsTest.java

```diff
@@ -186,4 +187,66 @@ public void similarTextSuggestionQuickFixForUndefinedObjectUsingLoopSectionMetad
 						ConfigurationItemEditType.update, "ignore", //
 						d)));
 	}
+
+	@Test
+	public void similarTextSuggestionQuickFixForUndefinedMethod() throws Exception {
+		String template = "{@java.lang.String string}\r\n" + //
+				"{string.charA()}";
+
+		Diagnostic d = d(1, 8, 1, 13, //
+				QuteErrorCode.UnknownMethod, //
+				"`charA` cannot be resolved or is not a method of `java.lang.String` Java type.", //
+				DiagnosticSeverity.Error);
+		d.setData(new JavaBaseTypeOfPartData("java.lang.String"));
+
+		testDiagnosticsFor(template, d);
+		testCodeActionsFor(template, d, //
+				ca(d, te(1, 8, 1, 13, "charAt")));
+	}
+
+	@Test
+	public void similarTextSuggestionQuickFixForUndefinedMethodWithResolver() throws Exception {
+		String template = "{@java.lang.String string}\r\n" + //
+				"{string.ifTru()}";
+
+		Diagnostic d = d(1, 8, 1, 13, //
+				QuteErrorCode.UnknownMethod, //
+				"`ifTru` cannot be resolved or is not a method of `java.lang.String` Java type.", //
+				DiagnosticSeverity.Error);
+		d.setData(new JavaBaseTypeOfPartData("java.lang.String"));
+
+		testDiagnosticsFor(template, d);
+		testCodeActionsFor(template, d, //
+				ca(d, te(1, 8, 1, 13, "ifTruthy")));
+	}
+
+	@Test
+	public void noSimilarTextSuggestionQuickFixForUndefinedMethod() throws Exception {
+		String template = "{@java.lang.String string}\r\n" + //
+				"{string.noMethod()}";
+
+		Diagnostic d = d(1, 8, 1, 16, //
+				QuteErrorCode.UnknownMethod, //
+				"`noMethod` cannot be resolved or is not a method of `java.lang.String` Java type.", //
+				DiagnosticSeverity.Error);
+		d.setData(new JavaBaseTypeOfPartData("java.lang.String"));
+
+		testDiagnosticsFor(template, d);
+		testCodeActionsFor(template, d);
+	}
+
+	@Test
+	public void noSimilarTextSuggestionQuickFixForUndefinedNamespaceMethod() throws Exception {
+		String template = "{cdi:bean.foo()}";
+
+		Diagnostic d = d(0, 10, 0, 13, //
+				QuteErrorCode.UnknownMethod, //
+				"`foo` cannot be resolved or is not a method of `java.lang.String` Java type.", //
+				DiagnosticSeverity.Error);
+		d.setData(new JavaBaseTypeOfPartData("java.lang.String"));
+
+		testDiagnosticsFor(template, d);
+		testCodeActionsFor(template, d);
+	}
+
```

Comment:
This test is not relevant, and the test name noSimilarTextSuggestionQuickFixForUndefinedNamespaceMethod has a wrong name since cdi namespace exists.

Please rename test with `similarTextSuggestionQuickFixForUndefinedMethodWithNamespace` The template to test should be:

`String template = "{cdi:bean.charA()}";`

and you should have a similar code action with `charAt`